libraries/rush-lib/src/logic/pnpm/SubspaceGlobalPnpmfileShim.ts

```diff
@@ -0,0 +1,166 @@
+    }
+    settings = context.splitWorkspacePnpmfileShimSettings!;
+  } else if (!context.splitWorkspacePnpmfileShimSettings) {
+    // Reuse the already initialized settings
+    context.splitWorkspacePnpmfileShimSettings = settings;
+  }
+  // If a userPnpmfilePath is provided, we expect it to exist
+  if (!userPnpmfile && settings.userPnpmfilePath) {
+    userPnpmfile = require(settings.userPnpmfilePath);
+  }
+  // If a semverPath is provided, we expect it to exist
+  if (!semver && settings.semverPath) {
+    semver = require(settings.semverPath);
+  }
+  if (!normalizePath && settings.pathNormalizerPath) {
+    normalizePath = require(settings.pathNormalizerPath);
+  }
+  // Return the normalized context
+  return context as IPnpmfileContext;
+}
+
+// Rewrite rush project referenced in split workspace.
+// For example: "project-a": "workspace:*" --> "project-a": "link:../../project-a"
+function rewriteRushProjectVersions(
+  packageName: string,
+  dependencies: { [dependencyName: string]: string } | undefined
+): void {
+  if (!dependencies) {
+    return;
+  }
+
+  if (!settings || !normalizePath) {
+    throw new Error(`splitWorkspaceGlobalPnpmfileShimSettings not initialized`);
+  }
+
+  const subspaceProject: IWorkspaceProjectInfo | undefined = settings.subspaceProjects[packageName];
+  if (!subspaceProject) {
+    return;
+  }
+
+  for (const dependencyName of Object.keys(dependencies)) {
+    const currentVersion: string = dependencies[dependencyName];
+
+    if (currentVersion.startsWith('workspace:')) {
+      const workspaceProjectInfo: IWorkspaceProjectInfo | undefined =
+        settings.workspaceProjects[dependencyName];
+      if (workspaceProjectInfo) {
+        // Case 1. "<package_name>": "workspace:*"
+        const relativePath: string = normalizePath(
+          path.relative(subspaceProject.projectRelativeFolder, workspaceProjectInfo.projectRelativeFolder)
+        );
+        const newVersion: string = 'link:' + relativePath;
+        dependencies[dependencyName] = newVersion;
+      } else {
+        // Case 2. "<alias>": "workspace:<aliased_package_name>@<version>"
+        const packageSpec: string = currentVersion.slice('workspace:'.length);
+        const nameEndsAt: number =
+          packageSpec[0] === '@' ? packageSpec.slice(1).indexOf('@') + 1 : packageSpec.indexOf('@');
+        const aliasedPackageName: string = nameEndsAt > 0 ? packageSpec.slice(0, nameEndsAt) : packageSpec;
+        // const depVersion: string = nameEndsAt > 0 ? packageSpec.slice(nameEndsAt + 1) : '';
+        const aliasedWorkspaceProjectInfo: IWorkspaceProjectInfo | undefined =
+          settings.workspaceProjects[aliasedPackageName];
+        if (aliasedWorkspaceProjectInfo) {
+          const relativePath: string = normalizePath(
+            path.relative(
+              subspaceProject.projectRelativeFolder,
+              aliasedWorkspaceProjectInfo.projectRelativeFolder
+            )
+          );
```

Comment:
Are you sure that we need to add this [normalize-path](https://www.npmjs.com/package/normalize-path) library?  (If the need for this operation hasn't arisen before, what is different about this operation?)

Maybe it is basically equivalent to [Path.convertToSlashes()](https://github.com/microsoft/rushstack/blob/a41647efd43841722009c124c0c64bacd0b7bcf2/libraries/node-core-library/src/Path.ts#L219)?

Reply:
Replaced with path.normalize()